### PR TITLE
pageserver: move `page_service`'s  `import basebackup` / `import wal` to mgmt API

### PR DESCRIPTION
**Pull request by problame**
I want to fix bugs in `page_service` ([issue](https://github.com/neondatabase/neon/issues/7427)) and the `import basebackup` / `import wal` stand in the way / make the refactoring more complicated.

We don't use these methods anyway in practice, but, there have been some objections to removing the functionality completely. 

So, this PR preserves the existing functionality but moves it into the HTTP management API.

Note that I don't try to fix existing bugs in the code, specifically not fixing
* it only ever worked correctly for unsharded tenants
* it doesn't clean up on error

All errors are mapped to `ApiError::InternalServerError`.

### CODE REVIEW

**Review by yliang412 — APPROVED** [review]
LGTM

**Comment by problame** [comment]
Waiting for review from team compute https://neondb.slack.com/archives/C04DGM6SMTM/p1720542945296569

**Review by hlinnaka — APPROVED** [review]
Fine by me. We might need to bring this back later, perhaps in a new HTTP-based API that's exposed to computes or something. But let's cross that bridge when we get there. In the meanwhile, it's good to keep the API surface as small as possible.

**Pull request merged**
